ovh/types_cloud_project_containerregistry_ip_restrictions.go

```diff
@@ -22,13 +22,15 @@ func (opts *CloudProjectContainerRegistryIPRestrictionCreateOpts) FromResource(d
 func loadIPRestrictionsFromResource(i interface{}) []CloudProjectContainerRegistryIPRestriction {
 	ips := make([]CloudProjectContainerRegistryIPRestriction, 0)
 
-	iprestrictionsSet := i.([]interface{})
-
-	for _, ipSet := range iprestrictionsSet {
-		ips = append(ips, CloudProjectContainerRegistryIPRestriction{
-			Description: ipSet.(map[string]interface{})["description"].(string),
-			IPBlock:     ipSet.(map[string]interface{})["ip_block"].(string),
-		})
+	iprestrictionsSet := i.(*schema.Set)
+
+	for _, ipSet := range iprestrictionsSet.List() {
+		if len(ipSet.(map[string]interface{})) > 0 {
```

Comment:
from what I see in the resources' schemas this can't happen since we verify that `ip_block` is set, so I guess we can remove this `if`. 
Side node, I'm not sure to understand why the elements in `ip_restrictions` are maps and not nested objects with mandatory fields, this could be updated.

Reply:
OK I was not aware of this issue, so keeping the `if` seems like a good idea indeed !
I want to avoid a breaking change so let's keep the resource as-is, and on new resources we won't have this problem as we use `terraform-plugin-framework`.